### PR TITLE
AeroAcoustics Model

### PR DESCRIPTION
**Pull request by ptrbortolotti**
**Complete this sentence**
THIS PULL REQUEST IS READY TO MERGE

**Feature or improvement description**
The aeroacoustics model is based on work performed over the past three decades. Frequency-based models for turbulent inflow; turbulent boundary layer – trailing edge; laminar boundary layer –vortex shedding; tip vortex; and trailing edge bluntness – vortex shedding noise mechanisms are included. A simple directivity model is also included.

**Related issue, if one exists**
None

**Impacted areas of the software**
AeroDyn15

**Additional supporting information**
An NREL technical will be available at this link https://www.nrel.gov/docs/fy20osti/75731.pdf and it is attached here
[75731_Noise_NREL_TR_200725.pdf](https://github.com/OpenFAST/openfast/files/4991295/75731_Noise_NREL_TR_200725.pdf)

**Test results, if applicable**
The r-test repo now includes test cases with the IEA Wind Task 37 land-based reference wind turbine

### CODE REVIEW

**Comment by andrew-platt** [comment]
Two minor items remaining:

- [ ] update VS projects
- [ ] run baselines for IEA turbine on windows-intel and linux-intel

File: share/docker/openfast_dev/Dockerfile [hunk 1]
```diff
@@ -42,4 +42,4 @@ RUN ctest -VV -L linear
 ## - 9, 16 because they're very sensitive
 ## - 19, 20 because theyre too long
 ## - 17, 22, 23 becuase we dont know why they fail :(
-RUN ctest -VV -j8 -I 1,1,1,2,3,4,5,6,7,8,10,11,12,13,14,15,18,21,24,25
```
Comment (rafmudaf) [inline share/docker/openfast_dev/Dockerfile:45]: FYI this does *not* enable running these tests in CI. For that, modify the shell script at `.github/actions/compile-and-test/entrypoint.sh`.

Comment (andrew-platt) [reply share/docker/openfast_dev/Dockerfile:45]: I'll make that change.

File: modules/aerodyn/src/AirfoilInfo.f90 [hunk 2]
```diff
@@ -477,9 +479,13 @@ SUBROUTINE ReadAFfile ( AFfile, NumCoefs, InCol_Alfa, InCol_Cl, InCol_Cd, InCol_
...
 
       ENDIF
-
-
-         ! How many columns do we need to read in the input and how many total coefficients will be used?
+      
+      ! Reading Boundary layer file  for aeroacoustics
+      CALL ParseVar ( FileInfo, CurLine, 'BL_file' , p%BL_file , ErrStat2, ErrMsg2, UnEc )
+         CALL SetErrStat( ErrStat2, ErrMsg2, ErrStat, ErrMsg, RoutineName )
+      p%BL_file=trim(PriPath)//trim(p%BL_file)
```
Comment (bjonkman) [inline modules/aerodyn/src/AirfoilInfo.f90:486]: Does this work as intended when `BL_file` is specified using an absolute path?

File: vs-build/AeroDyn/AeroDyn_Driver.vfproj [hunk 3]
```diff
@@ -290,38 +321,451 @@
...
 			<FileConfiguration Name="Release|x64">
-				<Tool Name="VFCustomBuildTool" CommandLine="CALL ..\RunRegistry.bat AeroDyn" Description="Running Registry for AeroDyn" Outputs="..\..\modules\AeroDyn\src\AeroDyn_Types.f90"/></FileConfiguration>
+				<Tool Name="VFFortranCompilerTool" StandardWarnings="standardWarningsNone"/></FileConfiguration>
 			<FileConfiguration Name="Debug|x64">
-				<Tool Name="VFCustomBuildTool" CommandLine="CALL ..\RunRegistry.bat AeroDyn" Description="Running Registry for AeroDyn" Outputs="..\..\modules\AeroDyn\src\AeroDyn_Types.f90"/></FileConfiguration>
+				<Tool Name="VFFortranCompilerTool" StandardWarnings="standardWarningsNone"/></FileConfiguration>
 			<FileConfiguration Name="Debug_Double|x64">
-				<Tool Name="VFCustomBuildTool" CommandLine="CALL ..\RunRegistry.bat AeroDyn" Description="Running Registry for AeroDyn" Outputs="..\..\modules\AeroDyn\src\AeroDyn_Types.f90"/></FileConfiguration>
+				<Tool Name="VFFortranCompilerTool" StandardWarnings="standardWarningsNone"/></FileConfiguration>
+			<FileConfiguration Name="Debug_Matlab|x64">
```
Comment (bjonkman) [inline vs-build/AeroDyn/AeroDyn_Driver.vfproj:342]: I see a few `FileConfiguration` lines in here for `Debug_Matlab` and `Release_Matlab`, but there those configurations don't exist for the AeroDyn driver Visual Studio project. Could you remove these extra lines?

File: vs-build/AeroDyn/AeroDyn_Driver.vfproj [hunk 4]
```diff
@@ -339,7 +783,20 @@
 				<Tool Name="VFFortranCompilerTool" WarnUnusedVariables="false"/></FileConfiguration></File>
 		<File RelativePath="..\..\modules\nwtc-library\src\NWTC_Num.f90"/>
 		<File RelativePath="..\..\modules\nwtc-library\src\SingPrec.f90"/>
-		<File RelativePath="..\..\modules\nwtc-library\src\SysIVF.f90"/></Filter>
+		<File RelativePath="..\..\modules\nwtc-library\src\SysIVF.f90">
+			<FileConfiguration Name="Debug_Matlab|x64" ExcludedFromBuild="true"/>
+			<FileConfiguration Name="Release_Matlab|x64" ExcludedFromBuild="true"/>
+			<FileConfiguration Name="Debug_Matlab|Win32" ExcludedFromBuild="true"/>
+			<FileConfiguration Name="Release_Matlab|Win32" ExcludedFromBuild="true"/></File>
+		<File RelativePath="..\..\modules\nwtc-library\src\SysMatlabWindows.f90">
```
Comment (bjonkman) [inline vs-build/AeroDyn/AeroDyn_Driver.vfproj:791]: Could you also remove the `SysMatlabWindows.f90` file from this project?

File: modules/aerodyn/src/AirfoilInfo.f90 (hunk 2 above)
Comment (andrew-platt) [reply modules/aerodyn/src/AirfoilInfo.f90:486]: Good question.  I'll take a look at that.

Comment (andrew-platt) [reply modules/aerodyn/src/AirfoilInfo.f90:486]: Confirmed your suspicion.  This does not work with absolute paths.  Fix in progress

File: vs-build/AeroDyn/AeroDyn_Driver.vfproj (hunk 3 above)
Comment (andrew-platt) [reply vs-build/AeroDyn/AeroDyn_Driver.vfproj:342]: Oops.  I'll remove those shortly.

File: vs-build/AeroDyn/AeroDyn_Driver.vfproj (hunk 4 above)
Comment (andrew-platt) [reply vs-build/AeroDyn/AeroDyn_Driver.vfproj]: done!

File: vs-build/AeroDyn/AeroDyn_Driver.vfproj (hunk 3 above)
Comment (andrew-platt) [reply vs-build/AeroDyn/AeroDyn_Driver.vfproj]: Done!

File: modules/aerodyn/src/AirfoilInfo.f90 (hunk 2 above)
Comment (andrew-platt) [reply modules/aerodyn/src/AirfoilInfo.f90]: Fixed.

**Comment by rafmudaf** [comment]
Is the r-test AeroAcoustics branch ready to be merged into dev, as well?

**Pull request merged**